monash-human-power/MHP_DAS
# Live Anemometer Data

## Description

Created Python scripts for wireless module and sensor base, where new sensors with scripts written in Python can be added to in the future.

Integrated the anemometer (wind sensor) into the data acquisition system (DAS). This allows the anemometer to collect wind speed and direction data, and publish these data to the MQTT broker under the topic `/v3/wireless_module/5/data`.


## Steps to Test
1. Install the [Vaisala USB Instrument Finder driver](https://go.vaisala.com/software/WXT530/Vaisala_WXT530_Configuration_Tool_Weather_Measurement.zip?_ga=2.138439603.1803271655.1674458831-1555859295.1674458831) using these [installation instructions](https://docs.vaisala.com/r/M211840EN-F/en-US/GUID-6D206CCD-21E9-4E9A-98C9-760C90EA90BF/GUID-AE6CDFA9-16A5-4E47-B354-37C04534C558).

2. Connect the anemometer to your computer and note down which port it is connecting to.

3. In the `wind_module` directory, create a local version of `config.py` using the `config.example.py` file. 

4. Get inside the `wind_module` directory using:
     ```
     cd wireless_mdoules_py/wind_module
    ```

5. Ensure that you have [poetry](https://python-poetry.org/) installed, spawn into the poetry environment using:
    ```
    poetry shell
    ```

6. Run the main program using:
    ```
    python main.py
    ```
    ![Running main program](https://user-images.githubusercontent.com/89284838/214009965-a6f4be07-f06f-4617-9eca-3625ac3289db.jpg)


7. Open 2 terminals:
    - One to subscribe to the topic `/v3/wireless_module/5/data`.
    - Another to publish to the topic `v3/start`.

8. Publish a true start message to the topic `v3/start`. Streams of data should be shown in the terminal subscribed to the topic `/v3/wireless_module/5/data`.

    ![Publishing data](https://user-images.githubusercontent.com/89284838/214011006-3dc4fe0e-c9e5-4654-904a-41759d833e2c.jpg)

9. While the data is being published, mess around with the anemometer to ensure values are changing accordingly.

10. Publish a false start message to the topic `v3/start`. The terminal subscribed to the topic `/v3/wireless_module/5/data` should look like this:

    ![Stop publishing data](https://user-images.githubusercontent.com/89284838/214012295-a1a9596e-70ac-4ecb-868e-636102f6cb23.jpg)
